crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -197,5 +202,11 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
 
             #debug_fn
         }
+
+        impl #impl_generics #bevy_reflect_path::serde::ReflectSerializableWithData for #struct_name #ty_generics #where_clause{
+            fn get_serialization_data() -> #bevy_reflect_path::serde::SerializationData {
+                #bevy_reflect_path::serde::SerializationData::new([#(#serialization_blacklist),*].into_iter())
+            }
+        }
```

Comment:
Theoretically, you don't need the trait. If you passed `Option<BitSet<u32>>` to `impl_get_registration` instead of a `bool`, then you could just construct the `SerializationData` there.

Or is there a reason for having a trait?

Reply:
No particular reason no, I'll remove the trait

Reply:
I can see it being a good idea for extending with other methods in the future, but with no current plans on what that might entail, yeah it might be good to remove it for now.

Reply:
I mostly just did it this way thinking that maybe accessing via trait might be useful, but then the serialisation module doesn't ever get access to raw types anyway, and I don't really see much need on the user side to ever touch this machinery